doc/man/man5/strbcm.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH STRBCM 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System File Formats"
.\"
.\"
.SH NAME
.B @PACKAGE@
\- @PACKAGE_TITLE@ package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@-@VERSION@ )
package for
.B "Linux Fast-STREAMS" .
.\"
.\"
.SH DESCRIPTION
.PP
The
.I @PACKAGE_TITLE@
package provides
.I STREAMS
modules and drivers, programs and daemons for the compilation
of externally supplied source and binary
.I STREAMS
modules.
.PP
The
.I @PACKAGE_TITLE@
package includes
kernel modules,
.I "SVR 4.2"
.I STREAMS
drivers, modules, libraries, utilities, test programs, daemons, and
development environment for the development and execution of
.I STREAMS
modules and drivers.  This package contains primarily a framework for the
compilation and linkage of externally supplied source and binary modules.
.PP
This distribution is only currently applicable to
.B Linux
2.4 and 2.6 kernels and was targetted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
architectures, but should build and install for other architectures as well.
.PP
This package is currently incomplete.  It is being released as a
reference point for the community.  If you are interested in completion
of this package, contact
.RB < info@openss7.com >.
.\"
.\"
.SH "PACKAGE CONTENTS"
.PP
The
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@-@VERSION@ )
package contains the following manual pages:
.HP
.BR tstdrv (4)\ \-\ \c
test driver.
.PD 0
.HP
.BR tstmod (4)\ \-\ \c
test module.
.HP
.BR strbcm (5)\ \-\ \c
@PACKAGE@-@VERSION@ package.
.HP
.BR strbcm_mknod (8)\ \-\ \c
create @PACKAGE@ devices.
.HP
.BR test-bcm (8)\ \-\ \c
test case executable.
.PD
.\"
.\"
.\".SH NOTICES
.\".PP
.\"
.\"
.\".SH EXAMPLES
.\".PP
.\"
.\"
.SH DEVICES
.PP
The
.BR testdrv (4)
driver.
.\"
.\"
.SH MODULES
.PP
The
.BR testmod (4)
module.
.\"
.\"
.\".SH FILES
.\".PP
.\"
.\"
.SH "SEE ALSO"
.PP
.BR tstdrv (4),
.BR tstmod (4),
.BR strbcm_mknod (8),
.BR test-bcm (8).
.\"
.\"
.\".SH COMPATIBILITY
.\".PP
.\"
.\"
.\".SH CONFORMANCE
.\".PP
.\"
.\"
.SH HISTORY
.PP
The
.B @PACKAGE@
package is part of OpenSS7
.B "Linux Fast-STREAMS" .
.[
LfS
.]
.\"
.\"
.[
$LIST$
.]
.TI


NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/util/init/strbcm.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strbcm:	start and stop strbcm modules
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strbcm
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		BCM character devices are present in the /dev directory and \
#		that the STREAMS BCM kernel modules are configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strbcm
# Required-Start: streams strcompat $network
# Required-Stop: streams strcompat $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS Binary Compatibility Modules
# License: GPL
# Description:	This STREAMS BCM init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS BCM character
#	devices are present in the /dev directory and that the STREAMS BCM
#	kernel modules are configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strbcm'
config="/etc/default/$name"
desc="the STREAMS BCM modules"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRBCM_MKNOD" ; then
    for STRBCM_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $STRBCM_MKNOD ] ; then
	    break
	else
	    STRBCM_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$STRBCM_PRELOAD"       ] || STRBCM_PRELOAD=""
[ -n "$STRBCM_DRIVERS"       ] || STRBCM_DRIVERS="streams-tstdrv"
[ -n "$STRBCM_MODULES"       ] || STRBCM_MODULES="streams-tstmod"
[ -n "$STRBCM_MAKEDEVICES"   ] || STRBCM_MAKEDEVICES="yes"
[ -n "$STRBCM_REMOVEDEVICES" ] || STRBCM_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRBCM_MKNOD" ] && STRBCM_MAKEDEVICES="no"
[ -z "$STRBCM_MKNOD" ] && STRBCM_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRBCM_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRBCM_MKNOD" -a :"$STRBCM_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS BCM devices: "
	$STRBCM_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRBCM_MKNOD" -a ":$STRBCM_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS devices: "
	$STRBCM_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRBCM_PRELOAD $STRBCM_DRIVERS $STRBCM_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.12  2008-10-21 22:42:14  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.11  2008-04-28 17:31:03  brian
# - updates for release
#
# Revision 0.9.2.10  2007/08/14 02:46:35  brian
# - GPLv3 headers
#
# Revision 0.9.2.9  2007/03/08 22:42:48  brian
# - correct redirect
#
# Revision 0.9.2.8  2007/03/08 22:12:43  brian
# - update headers
#
# Revision 0.9.2.7  2007/03/08 22:08:15  brian
# - sense of VERBOSE reversed
#
# Revision 0.9.2.6  2007/03/08 20:25:15  brian
# - ubuntu looks for lsb info in init scripts
#
# Revision 0.9.2.5  2006/10/16 00:21:16  brian
# - do not load too many kernel modules on init
#
# Revision 0.9.2.4  2006/10/13 06:59:51  brian
# - load drivers but not modules by default, but remove modules
#
# Revision 0.9.2.3  2006/10/13 03:59:53  brian
# - corrected init scripts and config files
#
# Revision 0.9.2.2  2006/08/16 07:40:46  brian
# - rework addition and removal of kernel modules
#
# Revision 0.9.2.1  2006/03/08 11:18:05  brian
# - completing package
#
# =============================================================================

# vim: ft=sh sw=4 tw=80
